tests/TestCase/ORM/TableTest.php

```diff
@@ -964,6 +964,137 @@ public function testDeleteAllFailure()
         $table->deleteAll(['id >' => 4]);
     }
 
+
+    /**
+     * @return array
+     */
+    public function dataDeleteEach()
+    {
+        return [
+            [['batch' => 1]],
+            [['batch' => 100]],
+            [['atomic' => false]],
+            [['stopOnFailure' => false]],
+            [['checkRules' => false]]
+        ];
+    }
+
+    /**
+     * @dataProvider dataDeleteEach
+     * @param array $options
+     */
+    public function testDeleteEach(array $options)
+    {
+        $table = new Table(['table' => 'users', 'connection' => $this->connection,]);
+
+        $beforeCount = 0;
+        $table->eventManager()->on('Model.beforeDeleteEach', function () use(&$beforeCount) {
+            $beforeCount++;
+        });
+
+        $afterCount = 0;
+        $table->eventManager()->on('Model.afterDeleteEach', function () use(&$afterCount) {
+            $afterCount++;
+        });
+
+        $result = $table->deleteEach(['username !=' => 'mariano'], $options);
+        $this->assertSame(3, $result);
+
+        if(isset($options['batch']) && $options['batch'] === 100) {
+            $this->assertSame(1, $beforeCount);
+            $this->assertSame(1, $afterCount);
+        } else {
+            $this->assertSame(3, $beforeCount);
+            $this->assertSame(3, $afterCount);
+        }
+
+        $result = $table->find('all')->toArray();
+        $this->assertCount(1, $result, 'Only one record should remain');
+        $this->assertEquals('mariano', $result[0]['username']);
+    }
+
+    /**
+     * @return array
+     */
+    public function dataDeleteEachInvalidArgument()
+    {
+        return [
+            [['limit' => 10]],
+            [['offset' => 10]],
+            [['page' => 10]]
+        ];
+    }
+
+    /**
+     * @param array $invalidParams
+     * @dataProvider dataDeleteEachInvalidArgument
+     * @expectedException \InvalidArgumentException
```

Comment:
You might also want to assert the message so you know the right exception case is being caught.